### PR TITLE
Add docs for new CCM tag merging feature

### PR DESCRIPTION
**Pull request by elihschiff**
### What does this PR do?

Adds docs for the new CCM tag merging feature

### Motivation

We want to help users better understand what happens when tags are merged from various sources

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by drichards-87** [comment]
@elihschiff Is this ready to be merged now?

**Comment by elihschiff** [comment]
Sorry this should be a draft PR. I will update it back to a regular PR once it is ready to merge. Thanks for checking

**Review by enemyghost — APPROVED** [review]

**Review by rtrieu — APPROVED** [review]
Looks good, thank you for this improvement! There's currently a code freeze, so I can't merge it right away. If you notice a long delay, please let us know in #documentation (Slack).

**Pull request merged**
